packages/common/src/utils/url-generator.ts

```diff
@@ -302,18 +289,61 @@ export const teamInviteLink = (inviteToken: string) =>
 
 export { dashboard };
 
-export const v2DefaultBranchUrl = (
-  owner: string,
-  name: string,
-  qsObject: Record<string, string> = {}
-) => {
-  const searchParams = new URLSearchParams({
-    ...qsObject,
-  });
+// This function handles all the scenarios of v2 branch editor urls
+// It is not exported from the package to avoid miss-using it
+const v2EditorBranchUrl = ({
+  owner,
+  repoName,
+  branchName,
+  workspaceId,
+  createDraftBranch,
+  source,
+}: {
+  owner: string;
+  repoName: string;
+  branchName?: string;
+  workspaceId?: string;
+  createDraftBranch?: boolean;
+  source?: string;
+}) => {
+  const queryString = new URLSearchParams({
+    ...(workspaceId ? { workspaceId } : {}),
+    ...(createDraftBranch ? { create: 'true' } : {}),
+    ...(source ? { utm_source: source } : {}),
+  }).toString();
```

Comment:
Do you think this can lead to maintainability issues as it grows?

Reply:
hopefully it won't grow a lot, even the workspaceId should ultimately be in the URL structure, not in the query string, this might be just a temp solution. As for the solution itself, i'd rather have explicit fields for things we want to add to the query then allowing dynamic objects to be serialized here. do you have any suggestions on how to improve this?